File: src/main/java/com/ftn/service/WarehouseCardAnalyticsService.java
package com.ftn.service;

import com.ftn.model.WarehouseCardAnalytics;
import com.ftn.model.enums.TrafficType;

import java.util.List;

/**
 * Created by Olivera on 31.5.2017..
 */
public interface WarehouseCardAnalyticsService {

    List<WarehouseCardAnalytics> read();

    WarehouseCardAnalytics read(Long id);

    List<WarehouseCardAnalytics> readByWarehouseCardId(Long warehouseCard);

    List<WarehouseCardAnalytics> read(Long id, TrafficType trafficType);

    WarehouseCardAnalytics create(WarehouseCardAnalytics warehouseCardAnalytics);

    WarehouseCardAnalytics update(Long id, WarehouseCardAnalytics warehouseCardAnalyticsDTO);

    void delete(Long id);
}

File: src/main/webapp/service/warehouseCardAnalyticsService.js
/**
 * Created by JELENA on 3.7.2017.
 */

app.service('warehouseCardAnalyticsService', function ($http) {
    return{
        read: function (warehouseCardId, onSuccess, onError) {
            $http.get('/api/warehouseCardAnalytics/warehouseCards/' + warehouseCardId).then(onSuccess, onError);
        }

    }
});
